Application/EdFi.Ods.Api/Security/Authorization/AuthorizationSegmentSqlProviderBase.cs

```diff
@@ -80,12 +79,54 @@ public AuthorizationSegmentSqlProviderBase(IAuthorizationViewsProvider authoriza
                     }
 
                     // Perform defensive checks against the remote possibility of SQL injection attack
-                    ValidateTableNameParts(claimEndpointName, subjectEndpointName, authorizationSegment.AuthorizationPathModifier);
+                    ValidateTableNameParts(
+                        claimEndpointName, subjectEndpointName, authorizationSegment.AuthorizationPathModifier);
 
-                    string derivedAuthorizationViewName = ViewNameHelper.GetFullyQualifiedAuthorizationViewName(
-                        subjectEndpointName,
-                        claimEndpointName,
-                        authorizationSegment.AuthorizationPathModifier);
+                    var entities = new List<string>()
```

Comment:
> The sub loop you're in here is intended to be after grouping by claims endpoints. The intent is that the claim endpoint name represents the data point you're authorizing as the source. In this case I think we can drastically simplify a lot of this by going to the code in AuthorizationBuilder, and change the GetClaimAuthorizationNameValuePairs method to build the Ed Org claims using "EducationOrganizationId" as the name, instead of relying on convention to generate a more specialized id property like LocalEducationAgencyId or SchoolId. That will cause our authorization to group up, so we do a check against the tuple table only once for any type of education organization.
> 
> Alternatively, if that's causing too much risk and pain to adjust the way the claims are built, you can group them up here once you determine they're EducationOrganizations. The goal should be we only hit the tuple table once and provide it a TVP with all the relevant ids in it, the same way it would have worked previously if you have more than one LocalEducationAgencyId to check for example.

 I tried to update this below line in GetClaimAuthorizationNameValuePairs method but problem is  AuthorizeSingleItemAsync method needs claimEducationOrganizationType to be LocalEducationAgency instead of EducationOrganizationId 
                  var claimNameValueTuple = Tuple.Create(
                        "EducationOrganizationId",
                        (object) educationOrganizationId);

Here in  AuthorizeSingleItemAsync method below line  is failing now  . Do I need to check this now if (inaccessibleIdentifierNames.Any(s => s.Equals(segment.SubjectEndpoint.Name))) ? 
var inaccessibleIdentifierNames = graph
                        .Vertices
                        .Except(graph.GetDescendantsOrSelf(claimEducationOrganizationType))
                        .Select(edOrgType => edOrgType + "Id") // NOTE: Embedded convention (adding Id suffix to EdOrg type)
                        .ToList();

Reply:
_claimsAuthorizationSegments add method where I know the AuthorizationSegmentEndpoint ,so  I am updating it in  ClaimsAuthorizationSegment class  in         public ClaimsAuthorizationSegment(
            IEnumerable<Tuple<string, object>> claimNamesAndValues,
            AuthorizationSegmentEndpoint subjectEndpoint,
            string authorizationPathModifier)